sdks/python/apache_beam/transforms/core.py

```diff
@@ -1294,16 +1295,43 @@ def _pardo_fn_data(self):
     windowing = None
     return self.fn, self.args, self.kwargs, si_tags_and_types, windowing
 
-  def to_runner_api_parameter(self, context):
+  def to_runner_api(self, context, main_inputs, has_parts=False):
```

Comment:
This is starting to look like a lot of code duplication. How about we pass (all) inputs as a keyword argument, and let `PTransform.to_runner_api` take an `**extra_kwargs` that it passes on to `to_runner_api_parameter`.